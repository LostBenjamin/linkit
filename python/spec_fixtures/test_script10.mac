# rainbow wandering pixels with mirror

%include symbols

$max <NUM-LEDS>
$max-half <max>
$render-time 50
$sequence-time 20
$fade-type ffd

$wander-range 3
$wander-neg-range `-(<wander-range> / 2)`

$brightness 20

[fancy 10]
    (init)
    (setup-rendering)
    (start-components)

[init]
    app
    <brightness>:lev

[setup-rendering]
    <render-time>,<render>:sch

[render !]
    flu

[start-components]
    (start-RED)
    (start-ORANGE)
    (start-YELLOW)
    (start-GREEN)
    (start-BLUE)
    (start-PURPLE)

((wanderer RED))
((wanderer ORANGE))
((wanderer YELLOW))
((wanderer GREEN))
((wanderer BLUE))
((wanderer PURPLE))

[[wanderer X
    [start-X]
        {seq-X},<max>,0:seq
        <sequence-time>,<sequence-X>:sch

    [sequence-X]
        <seq-X>,<MACRO-SEQ>,<math-X>:seq
        pos
        <seq-X>,0,<ADD-COLOR>:pal
        <fade-type>
        rst

    [math-X]
        <wander-range>:rng:psh
        <wander-neg-range>:psh
        add
        <seq-X>,<GET-CURRENT>:seq:psh
        add
        <max-half>:psh
        mod
]]

