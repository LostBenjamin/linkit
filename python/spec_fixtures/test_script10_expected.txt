10:set:25:run:28:run:32:run
14:set:12,90,0:seq:20,18:sch
15:set:3:rng:psh:-1:psh:add:13,-1:seq:psh:add:90:psh:mod
16:set:11,-4,21:seq:pos:11,0,1:pal:ffd:rst
17:set:14,-4,30:seq:pos:14,0,1:pal:ffd:rst
18:set:12,-4,19:seq:pos:12,0,1:pal:ffd:rst
19:set:3:rng:psh:-1:psh:add:12,-1:seq:psh:add:90:psh:mod
20:set:13,90,0:seq:20,31:sch
21:set:3:rng:psh:-1:psh:add:11,-1:seq:psh:add:90:psh:mod
22:set:11,90,0:seq:20,16:sch
23:set:3:rng:psh:-1:psh:add:10,-1:seq:psh:add:90:psh:mod
24:set:10,-4,23:seq:pos:10,0,1:pal:ffd:rst
25:set:app:20:lev
26:set:15,-4,27:seq:pos:15,0,1:pal:ffd:rst
27:set:3:rng:psh:-1:psh:add:15,-1:seq:psh:add:90:psh:mod
28:set:50,50:sch
29:set:14,90,0:seq:20,17:sch
30:set:3:rng:psh:-1:psh:add:14,-1:seq:psh:add:90:psh:mod
31:set:13,-4,15:seq:pos:13,0,1:pal:ffd:rst
32:set:34:run:22:run:14:run:20:run:29:run:33:run
33:set:15,90,0:seq:20,26:sch
34:set:10,90,0:seq:20,24:sch
50:set:flu

